looprpc/server.proto

```diff
@@ -177,3 +206,76 @@ message ServerLoopOutPushPreimageRequest {
 
 message ServerLoopOutPushPreimageResponse {
 }
+
+message SubscribeUpdatesRequest {
+    // The protocol version that the client adheres to.
+    ProtocolVersion protocol_version = 1;
+
+    // Swap hash is the hash of the swap to subscribe to updates for.
+    bytes swap_hash = 2;
+}
+
+// ServerSwapState is an enum which represents all the states a swap may have
+// from the server's perspective.
+enum ServerSwapState {
+    // The server has created the swap.
+    INITIATED = 0;
+
+    // The server has published the loop out on chain htlc.
+    HTLC_PUBLISHED = 1;
+
+    // The swap completed successfully.
+    SUCCESS = 2;
+
+    /*
+    The swap failed for a reason that is unknown to the server, this is only
+    set for older swaps.
+    */
+    FAILED_UNKNOWN = 3;
+
+    // No htlc was confirmed in time for the loop in swap to complete.
+    FAILED_NO_HTLC = 4;
+
+    // A loop in htlc confirmed on chain, but it did not have the correct value.
+    FAILED_INVALID_HTLC_AMOUNT = 5;
+
+    /*
+    We did not succeed in completing the loop in off chain payment before the
+    timeout.
+    */
+    FAILED_OFF_CHAIN_TIMEOUT = 6;
+
+    // The on chain timeout was claimed.
+    FAILED_TIMEOUT = 7;
+
+    /*
+    The server could not publish the loop out on chain htlc before the deadline
+    provided.
+    */
+    FAILED_SWAP_DEADLINE = 8;
+
+    // The server could not publish the loop out on chain htlc.
+    FAILED_HTLC_PUBLICATION = 9;
+
+    // The server has published the loop out on chain timeout tx.
+    TIMEOUT_PUBLISHED = 10;
+
+    // The swap has failed for unknown reasons, it will not be completed.
+    UNEXPECTED_FAILURE = 11;
```

Comment:
If we are keeping these, need the htlc confirmed state that was just added.

Reply:
added